tests/ha/control_pod/test_control_pod_soft_failures.py

```diff
@@ -0,0 +1,617 @@
+
+        LOGGER.info("Adding %s delay in config files of all control pods",
+                    self.min_proc_restart_delay)
+        self.add_proc_restart_delay(self.min_proc_restart_delay)
+        LOGGER.info("Successfully added %s delay in config files of all control pods",
+                    self.min_proc_restart_delay)
+        self.change_proc_delay = True
+
+        LOGGER.info("Step 1: Perform WRITE-READ-Verify on user %s", self.uids[0])
+        self.test_prefix = 'test-45500'
+        resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(self.users.values())[0],
+                                                    log_prefix=self.test_prefix, skipcleanup=True)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 1: Performed WRITEs-READs-Verify with variable sizes objects.")
+
+        LOGGER.info("Create %s iam users for deletion", num_users)
+        del_users = self.mgnt_ops.create_account_users(nusers=num_users)
+
+        LOGGER.info("Step 2: Perform IAM user creation/deletion in background")
+        args = {'user_crud': True, 'num_users': num_users, 'output': iam_output,
+                'del_users_dict': del_users, 'header': self.header}
+        thread1 = threading.Thread(target=self.ha_obj.iam_bucket_cruds, args=(event,), kwargs=args)
+        thread1.daemon = True  # Daemonize thread
+
+        args = {'user_crud': False, 'bkt_crud': True, 'num_bkts': num_bkts, 's3_obj': self.s3_obj,
+                'output': bkt_output}
+        thread2 = threading.Thread(target=self.ha_obj.iam_bucket_cruds, args=(event,), kwargs=args)
+        thread2.daemon = True  # Daemonize thread
+        thread1.start()
+        thread2.start()
+        LOGGER.info("Step 2: Successfully started IAM user and bucket creation/deletion in "
+                    "background")
+        LOGGER.info("Waiting for %s sec for background operations to start",
+                    HA_CFG["common_params"]["10sec_delay"])
+        time.sleep(HA_CFG["common_params"]["10sec_delay"])
+
+        LOGGER.info("Step 3: Create soft failures")
+        event.set()
+        for pod in self.ctrl_pod_list:
+            LOGGER.info("Creating soft failure on pod %s", pod)
+            resp = self.node_master_list[0].kill_process_in_container(
+                pod_name=pod, container_name=self.container_name, pid=self.pid, safe_kill=True)
+            LOGGER.info("Response: %s", resp)
+        LOGGER.info("Step 3: Created soft failures")
+
+        LOGGER.info("Waiting for %s time for all control pod containers to start",
+                    self.min_proc_restart_delay)
+        time.sleep(self.min_proc_restart_delay)
+        event.clear()
+
+        LOGGER.info("Waiting for threads to join")
+        thread1.join()
+        thread2.join()
+
+        LOGGER.info("Step 4: Checking cluster status")
+        resp = self.ha_obj.poll_cluster_status(self.node_master_list[0], timeout=180)
```

Comment:
again can we not check cluster status directly here as we have already waited for proc restart delay? or maybe timeout can be just delay?